pkg/puller/puller.go

```diff
@@ -390,13 +283,30 @@ func (p *Puller) histSyncWorker(ctx context.Context, peer swarm.Address, bin uin
 }
 
 func (p *Puller) liveSyncWorker(ctx context.Context, peer swarm.Address, bin uint8, cur uint64) {
-	loggerV2 := p.logger.V(2).Register()
+	loggerV2 := p.logger
 
 	defer p.wg.Done()
 	loggerV2.Debug("liveSyncWorker starting", "peer_address", peer, "bin", bin, "cursor", cur)
 	from := cur + 1
+
+	sleep := false
+
 	for {
 		p.metrics.LiveWorkerIterCounter.Inc()
+
+		if sleep {
+			select {
+			case <-p.quit:
+				loggerV2.Debug("liveSyncWorker quit on shutdown", "peer_address", peer, "bin", bin, "cursor", cur)
+				return
+			case <-ctx.Done():
+				loggerV2.Debug("liveSyncWorker context cancelled", "peer_address", peer, "bin", bin, "cursor", cur)
+				return
```

Comment:
Should `ctx.Done()` signal be enough, as `p.quity` seems to be canceling context? (There are multiple occurrences of this pattern)

Anyhow, it would be simpler to follow if there was only one end signal. Preferably create one `Context` and use that context as ending signal propagate where necessary).

Reply:
will look into this, thanks.

Reply:
we have a global context that is created in manage, it's used to signal blocking processes to terminate on exit. 
We also need a perBin contexts in case we need to stop syncing a specific bin.

Reply:
If perBin context is created using global context, then canceling global context would cancel perBin context. If context are created in this way then `p.quit` is not needed.

Reply:
i see what you mean now, we do not need to listen on p.quite in the syncing for loops. Thanks.